File: Turquoise/UI/TQLogoutConfirmationView.m
#import "TQLogoutConfirmationView.h"

#import "TQUserInfoManager.h"

@implementation TQLogoutConfirmationView

- (instancetype)init {
  self = (TQLogoutConfirmationView *) [UIView tq_loadFrom:@"TQLogoutConfirmationView" owner:self];
  return self;
}

- (IBAction)cancelButtonDidTap:(id)sender {
  [[TQOverlay sharedInstance] dismissWithAnimated:YES];
}

- (IBAction)logoutButtonDidTap:(id)sender {
  [[TQUserInfoManager sharedInstance] resetUserCredentials];
  [[TQOverlay sharedInstance] dismissWithAnimated:YES];
}

@end

File: Turquoise/UI/TQLogoutConfirmationView.swift
import Foundation
import UIKit

class TQLogoutConfirmationView : UIView {
  class func loadFromNib() -> TQLogoutConfirmationView? {
    return UIView.tq_load(from: "TQLogoutConfirmationView", owner: self) as? TQLogoutConfirmationView
  }

  @IBAction func cancelButtonDidTap(_ sender: Any?) {
    TQOverlay.sharedInstance.dismiss(animated: true)
  }

  @IBAction func logoutButtonDidTap(_ sender: Any?) {
    TQUserInfoManager.sharedInstance.resetUserCredentials()
    TQOverlay.sharedInstance.dismiss(animated: true)
  }
}
